### PR TITLE
refactor of item search, removal of unwanted methods

### PR DESCRIPTION
**Pull request by rpgayatri**
refactor of item search, removal of unwanted methods

### CODE REVIEW

File: src/main/java/life/genny/utils/PaymentUtils.java [hunk 1]
```diff
@@ -803,17 +788,17 @@ public static void updateUserPhoneNumber(BaseEntity userBe, String assemblyUserI
...
-			Double itemPrice = (Double) itemDescObj.get("amount");
+			String itemResponse = PaymentEndpoint.getPaymentItem(itemId, assemblyAuthKey);
+			
+			/* convert string into item-search object */
+			QPaymentsAssemblyItemSearchResponse itemObj = JsonUtils.fromJson(itemResponse, QPaymentsAssemblyItemSearchResponse.class);
+			System.out.println("item object ::"+itemObj);
+			
+			QPaymentsAssemblyItemResponse items = itemObj.getItems();
+			
+			Double itemPrice = items.getAmount();
```
Comment (DevJS) [inline src/main/java/life/genny/utils/PaymentUtils.java]: what if items is null ?

File: src/main/java/life/genny/utils/PaymentUtils.java [hunk 2]
```diff
@@ -829,23 +814,26 @@ public static Boolean checkForAssemblyItemValidity(String itemId, BaseEntity off
 			/* convert into cents */
 			System.out.println("calculated item price in cents ::"+calculatedItemPriceInCents);
 
-			Map<String, Object> buyerOwnerInfo = (Map<String, Object>) itemDescObj.get("buyer");
-			Map<String, Object> ownerContactInfo = (Map<String, Object>) buyerOwnerInfo.get("contactInfo");
+			QPaymentsAssemblyUser buyerOwnerInfo = items.getBuyer();
```
Comment (DevJS) [inline src/main/java/life/genny/utils/PaymentUtils.java]: nice

File: src/main/java/life/genny/utils/PaymentUtils.java [hunk 3]
```diff
@@ -829,23 +814,26 @@ public static Boolean checkForAssemblyItemValidity(String itemId, BaseEntity off
...
+			QPaymentsUserContactInfo ownerContactInfo = buyerOwnerInfo.getContactInfo();
```
Comment (DevJS) [inline src/main/java/life/genny/utils/PaymentUtils.java]: what if buyer is null ?

File: src/main/java/life/genny/utils/PaymentUtils.java [hunk 4]
```diff
@@ -829,23 +814,26 @@ public static Boolean checkForAssemblyItemValidity(String itemId, BaseEntity off
...
 
-			Map<String, Object> sellerDriverInfo = (Map<String, Object>) itemDescObj.get("seller");
-			Map<String, Object> driverContactInfo = (Map<String, Object>) sellerDriverInfo.get("contactInfo");
+			QPaymentsAssemblyUser sellerDriverInfo = items.getSeller();
```
Comment (DevJS) [inline src/main/java/life/genny/utils/PaymentUtils.java]: what if seller is null?

File: src/main/java/life/genny/utils/PaymentUtils.java [hunk 5]
```diff
@@ -829,23 +814,26 @@ public static Boolean checkForAssemblyItemValidity(String itemId, BaseEntity off
...
+			QPaymentsUserContactInfo driverContactInfo = sellerDriverInfo.getContactInfo();
 
-			Boolean isOwnerEmail = ownerContactInfo.get("email").equals(ownerEmail);
+			/* compare if item-owner is same as offer-owner */
+			Boolean isOwnerEmail = ownerContactInfo.getEmail().equals(ownerEmail);
```
Comment (DevJS) [inline src/main/java/life/genny/utils/PaymentUtils.java]: what if email is null ?

**Comment by rpgayatri** [comment]
all null checks provided.

**Review by DevJS — APPROVED** [review]
ok

**Review by DevJS — APPROVED** [review]
ok

**Pull request merged**